transmart-api-server/src/main/groovy/org/transmartproject/api/server/client/CustomClientRequestFactory.groovy

```diff
@@ -0,0 +1,64 @@
+package org.transmartproject.api.server.client
+
+import org.apache.http.client.methods.HttpUriRequest
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.annotation.Scope
+import org.springframework.http.*
+import org.springframework.http.client.ClientHttpRequestFactory
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory
+import org.springframework.stereotype.Component
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+import org.springframework.web.client.RestTemplate
+
+@Component
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+class CustomClientRequestFactory extends HttpComponentsClientHttpRequestFactory implements ClientHttpRequestFactory {
+
+    @Value('${keycloak.clientId}')
+    private String clientId
+
+    @Value('${keycloak.offlineToken}')
+    private String offlineToken
+
+    @Value('${keycloak.realm}')
+    private String realm
+
+    @Value('${keycloak.serverUrl}')
+    private String keycloakServerUrl
+
+    public static final String AUTHORIZATION_HEADER = "Authorization"
+
+
+    @Override
+    protected void postProcessHttpRequest(HttpUriRequest request) {
+        String offlineToken = getOfflineAccessToken()
```

Comment:
it's not the offline token. We might apply some caching here at some point: fetch new access token only when the current one has expired. For 2 rest api calls it's not that important though.